src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs

```diff
@@ -116,21 +116,41 @@ public ModelTypeProviderFields(InputModelType inputModel, TypeFactory typeFactor
         public IEnumerator<FieldDeclaration> GetEnumerator() => _fields.GetEnumerator();
         IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
 
+        private static bool ShouldPropertyOmitSetter(InputModelType inputModel, InputModelProperty property, CSharpType type)
+        {
+            if (property.IsDiscriminator)
+            {
+                // Discriminator properties should be writeable
+                return false;
```

Comment:
Could you explain why we need the discriminator property to be writeable?

Reply:
Sure